awx_collection/plugins/modules/tower_notification.py

```diff
@@ -26,168 +26,188 @@
     name:
       description:
         - The name of the notification.
+      type: str
       required: True
+    new_name:
+      description:
+        - Setting this option will change the existing name (looked up via the name field.
       type: str
     description:
       description:
         - The description of the notification.
-      required: False
       type: str
     organization:
       description:
         - The organization the notification belongs to.
-      required: False
       type: str
     notification_type:
       description:
         - The type of notification to be sent.
       required: True
-      choices: ["email", "slack", "twilio", "pagerduty", "hipchat", "webhook", "irc"]
+      choices:
+        - 'email'
+        - 'grafana'
+        - 'hipchat'
+        - 'irc'
+        - 'mattermost'
+        - 'pagerduty'
+        - 'rocketchat'
+        - 'slack'
+        - 'twilio'
+        - 'webhook'
       type: str
     notification_configuration:
       description:
         - The notification configuration file. Note providing this field would disable all notification-configuration-related fields.
-      required: False
-      type: str
+      type: dict
```

Comment:
The type of `notification_configuration` was changed to dict here. This is the right thing to do, my only request here is that we need to talk out the usage implications in `description`

 - This field used to take a file location, use Ansible `lookup` for these cases, this should go in the README.md backward-incompatible changes
 - This is the _preferred_ way of using the module, replacing the deprecated individual fields
 - If a property is provided via a deprecated field and inside this dict, then the value in this dict will take effect

I truly don't think we can work around the backward-incompatible change without the behavior being crazy nonsense. Making this a dict gives compatibility with the API.

Reply:
Updated readme.